Spigot-Server-Patches/0416-Add-damaged-slot-to-PlayerItemDamageEvent.patch

```diff
@@ -0,0 +1,41 @@
+From 90c6f3575a8d8b003dd04f795160736df4d068a2 Mon Sep 17 00:00:00 2001
+From: MTM123 <matiss234@gmail.com>
+Date: Mon, 4 Feb 2019 23:06:32 +0200
+Subject: [PATCH] Add damaged slot to PlayerItemDamageEvent
+
+
+diff --git a/src/main/java/net/minecraft/server/ItemStack.java b/src/main/java/net/minecraft/server/ItemStack.java
+index 7827f692..c06adf40 100644
+--- a/src/main/java/net/minecraft/server/ItemStack.java
++++ b/src/main/java/net/minecraft/server/ItemStack.java
+@@ -395,7 +395,26 @@ public final class ItemStack {
+                 i -= k;
+                 // CraftBukkit start
+                 if (entityplayer != null) {
+-                    PlayerItemDamageEvent event = new PlayerItemDamageEvent(entityplayer.getBukkitEntity(), CraftItemStack.asCraftMirror(this), i);
++
++                    //Paper start - getting slot that was damaged
++                    int slot = -1;
++                    if(entityplayer.getItemInMainHand() == this){
++                        slot = entityplayer.inventory.itemInHandIndex;
++                    }else if(entityplayer.getItemInOffHand() == this){
++                        slot = 104;
```

Comment:
Instead of hardcoding this slot ID, is there a better way to get it e.g. from an enum?